docs/how-to-guides/azure-arc-onboarding.md

```diff
@@ -0,0 +1,223 @@
+
+   c. Provide Azure Arc related permissions to the Service principal object(App registration). [Azure
+documentation](https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal)
+      provides a guided way to add permissions to the Service principal.
+
+   The following set of roles are needed
+     * Kubernetes Cluster - Azure Arc Onboarding
+     * Microsoft.Kubernetes connected cluster role
+     * Owner
+
+## Steps
+
+### Step 1: Configure [azure-arc-onboarding](../configuration-reference/components/azure-arc-onboarding.md)
+Lokomotive component.
+
+#### Config
+
+Create a file named `azure-arc-onboarding.lokocfg` with the following contents:
+
+```tf
+component "azure-arc-onboarding" {
+  application_client_id = "29348jdw-9g23-9kot-21sa-opw129831c2k"
+  application_password  = "foobar"
+  tenant_id             = "s38kjs4k-x123-89h2-7f21-89uffo109921"
+  resource_group        = "AzureArcTest"
+  cluster_name          = "mercury"
+}
+```
+
+Visit the component's [configuration reference](../configuration-reference/components/rook.md) for more
+information.
+
+#### Deploy the component
+
+Execute the following command to deploy the `azure-arc-onboarding` component:
+
+```bash
+lokoctl component apply azure-arc-onboarding
+```
+
+Verify the pod in the `azure-arc-onboarding` namespace is in the `Completed` state (this may take a few minutes):
+
+```console
+$ kubectl -n rook get pods -n azure-arc-onboarding
+NAME                       READY   STATUS      RESTARTS   AGE
+azure-arc-register-4s54j   0/1     Completed   0          81m
+```
+
+Azure Arc installs varios Helm charts in the `azure-arc` namespace, verify that all the deployments are in
+`Running` state:
+
+```console
+$ kubectl -n azure-arc get pods
+NAME                                         READY   STATUS    RESTARTS   AGE
+cluster-metadata-operator-7cff574c4f-ld8cv   2/2     Running   0          83m
+clusterconnect-agent-6dfd867c68-629xg        3/3     Running   0          83m
+clusteridentityoperator-fd498bf96-dqlhr      2/2     Running   0          83m
+config-agent-bd647558d-f46xn                 2/2     Running   0          83m
+controller-manager-8676dcdc6-mv68p           2/2     Running   0          83m
+extension-manager-bcfd5b597-lsk85            2/2     Running   0          83m
+flux-logs-agent-6596f58c56-55l7r             1/1     Running   0          83m
+kube-aad-proxy-97bf4bcf8-g9djl               2/2     Running   0          83m
+metrics-agent-5b9b94754f-qf5q2               2/2     Running   0          83m
+resource-sync-agent-f8c7c6b6b-zvhx4          2/2     Running   0          83m
+```
+
+### Step 3: Create a Kubernetes configuration
+
+#### Install `k8s-configuration` Azure CLI extension
+
+In order to create a Kubernetes configuration, we need to first install the `k8s-configurtion` Azure CLi
+extension:
+
+```bash
+az extension add --name k8s-configuration
+```
+
+#### Create a configuration
+
+```bash
```

Comment:
I'd add a word of explanation here, like (suggestion is more like a placeholder):
```suggestion
#### Create a configuration

Now, we can create example configuration by executing the command below:

```bash
```